### PR TITLE
feat: Add a cast feature

### PR DESCRIPTION
**Pull request by lovenoble**

### CODE REVIEW

File: fhevm/tee_cast_test.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+		{"Tee16Cast32", tfhe.FheUint16, tfhe.FheUint32, 4283, 4283},
+		{"Tee16Cast64", tfhe.FheUint16, tfhe.FheUint64, 4283, 4283},
+		{"Tee32Cast4", tfhe.FheUint32, tfhe.FheUint4, 1333337, 9},
+		{"Tee32Cast8", tfhe.FheUint32, tfhe.FheUint8, 1333337, 89},
+		{"Tee32Cast16", tfhe.FheUint32, tfhe.FheUint16, 1333337, 22617},
+		{"Tee32Cast64", tfhe.FheUint32, tfhe.FheUint64, 1333337, 1333337},
+		{"Tee64Cast4", tfhe.FheUint64, tfhe.FheUint4, 13333377777777777, 1},
+		{"Tee64Cast8", tfhe.FheUint64, tfhe.FheUint8, 13333377777777777, 113},
+		{"Tee64Cast16", tfhe.FheUint64, tfhe.FheUint16, 13333377777777777, 48241},
+		{"Tee64Cast32", tfhe.FheUint64, tfhe.FheUint32, 13333377777777777, 3994664049},
```
Comment (amaury1093) [inline fhevm/tee_cast_test.go:39]: Shouldn't this error?

How does this work currently using TFHE, for example what happens if I do:
```
euint32 a = TFHE.asEuint32(<big number>);
euint16 b = TFHE.asEuint16(a);
```

Comment (lovenoble) [reply fhevm/tee_cast_test.go:39]: b_plain = a_plain mod (2^16)

**Comment by lovenoble** [comment]
There was the bug in unmarshalling fhe type value so I created a new PR https://github.com/Inco-fhevm/fhevm-go/pull/26 to fix it.

**Comment by amaury1093** [comment]
Why was this PR merged in the first place @lovenoble? It wasn't approved yet

**Pull request merged**